ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+            {
+                case 'C':
+                case 'L':
+                    memcpy(whites, record + l_glop1,
+                           (canvi_amplada < 0
+                                ? nNewWidth
+                                : data_base_XP->pField[nIField].BytesPerField));
+                    retorn_fwrite = fwrite_function(whites, nNewWidth, 1,
+                                                    data_base_XP->pfDataBase);
+
+                    if (1 != retorn_fwrite)
+                    {
+                        if (whites)
+                            free_function(whites);
+                        free_function(record);
+                        return 1;
+                    }
+                    break;
+                case 'N':
+                    if (nNewPrecision ==
+                            data_base_XP->pField[nIField].DecimalsIfFloat ||
+                        que_fer_amb_reformatat_decimals ==
+                            MM_NOU_N_DECIMALS_NO_APLICA)
+                        que_fer_amb_reformatat_decimals =
+                            MM_NOMES_DOCUMENTAR_NOU_N_DECIMALS;
+                    else if (que_fer_amb_reformatat_decimals ==
+                             MM_PREGUNTA_SI_APLICAR_NOU_N_DECIM)
+                        que_fer_amb_reformatat_decimals =
+                            MM_NOMES_DOCUMENTAR_NOU_N_DECIMALS;
+
+                    if (que_fer_amb_reformatat_decimals ==
+                        MM_NOMES_DOCUMENTAR_NOU_N_DECIMALS)
+                    {
+                        if (canvi_amplada >= 0)
+                        {
+                            if (1 !=
+                                    fwrite_function(whites, canvi_amplada, 1,
+                                                    data_base_XP->pfDataBase) ||
+                                1 != fwrite_function(
+                                         record + l_glop1,
+                                         data_base_XP->pField[nIField]
+                                             .BytesPerField,
+                                         1, data_base_XP->pfDataBase))
+                            {
+                                if (whites)
+                                    free_function(whites);
+                                free_function(record);
+                                return 1;
+                            }
+                        }
+                        else if (canvi_amplada < 0)
+                        {
+
+#ifdef _MSC_VER
+#pragma warning(disable : 4127)
+#endif
+                            for (j = (GInt32)(l_glop1 +
+                                              (data_base_XP->pField[nIField]
+                                                   .BytesPerField -
+                                               1));
+                                 TRUE; j--)
+#ifdef _MSC_VER
+#pragma warning(default : 4127)
+#endif
+                            {
```

Comment:
perhaps ?
```suggestion
                            j = (GInt32)(l_glop1 +
                                              (data_base_XP->pField[nIField]
                                                   .BytesPerField));    // Note that I've remove the - 1 here, as it will be applied in the first iteration below
                            while (TRUE)
                            {
                                --j;
```

Reply:
Done